### PR TITLE
GAWB-3133: health ("helth") monitor

### PR DESCRIPTION
**Pull request by davidangb**
* import the `HealthMonitor` from workbench-libs and use it to power status checks
* break `StatusService` into its own class, get the status handler + route out of `AgoraService`
* beef up unit tests for the status route, including un-ignoring the healthy test case
* remove now-unused `AgoraStatus` model object, in deference to workbench-libs' `StatusCheckResponse`
* correct an instance of `it's` to the proper `its`; ironic given the typo in my branch name and extraneous trailing colon in my commit message

See also:
broadinstitute/firecloud-orchestration#600
broadinstitute/rawls#853

to update consumers of this status endpoint.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by davidangb** [comment]
https://fc-jenkins.dsp-techops.broadinstitute.org/job/firecloud-ui-fiab-test-runner/56/consoleFull
```
[info] *** 9 TESTS FAILED ***
[error] Failed: Total 116, Failed 9, Errors 0, Passed 107, Ignored 7
[error] Failed tests:
[error] 	org.broadinstitute.dsde.firecloud.test.api.orch.OrchestrationApiSpec
[error] 	org.broadinstitute.dsde.firecloud.test.metadata.DataSpec
[error] (test:testOnly) sbt.TestsFailedException: Tests unsuccessful

[info] DataSpec:
[info] Download should reflect visible columns
[info] - no workspace defaults or user preferences *** FAILED ***
[info]   org.openqa.selenium.TimeoutException: Expected condition failed: waiting for org.broadinstitute.dsde.workbench.service.test.WebBrowserUtil$await$$anon$1@799e2fa3 (tried for 30 second(s) with 500 MILLISECONDS interval)
[info] Build info: version: '3.8.1', revision: '6e95a6684b', time: '2017-12-01T18:33:54.468Z'

[info] - no workspace defaults, with user preferences *** FAILED ***
[info]   org.openqa.selenium.TimeoutException: Expected condition failed: waiting for org.broadinstitute.dsde.workbench.service.test.WebBrowserUtil$await$$anon$1@129c9ced (tried for 30 second(s) with 500 MILLISECONDS interval)
[info] Build info: version: '3.8.1', revision: '6e95a6684b', time: '2017-12-01T18:33:54.468Z'

[info] - no workspace defaults, with user preferences *** FAILED ***
[info]   org.openqa.selenium.TimeoutException: Expected condition failed: waiting for org.broadinstitute.dsde.workbench.service.test.WebBrowserUtil$await$$anon$1@129c9ced (tried for 30 second(s) with 500 MILLISECONDS interval)
[info] Build info: version: '3.8.1', revision: '6e95a6684b', time: '2017-12-01T18:33:54.468Z'

[info] - no workspace defaults, with user preferences, new columns *** FAILED ***
[info]   org.openqa.selenium.TimeoutException: Expected condition failed: waiting for org.broadinstitute.dsde.workbench.service.test.WebBrowserUtil$await$$anon$1@75648466 (tried for 30 second(s) with 500 MILLISECONDS interval)
[info] Build info: version: '3.8.1', revision: '6e95a6684b', time: '2017-12-01T18:33:54.468Z'

[info] - with workspace defaults, no user preferences *** FAILED ***
[info]   org.openqa.selenium.TimeoutException: Expected condition failed: waiting for org.broadinstitute.dsde.workbench.service.test.WebBrowserUtil$await$$anon$1@7288eaac (tried for 30 second(s) with 500 MILLISECONDS interval)
[info] Build info: version: '3.8.1', revision: '6e95a6684b', time: '2017-12-01T18:33:54.468Z'

[info] - with workspace defaults, no user preferences, new columns *** FAILED ***
[info]   org.openqa.selenium.TimeoutException: Expected condition failed: waiting for org.broadinstitute.dsde.workbench.service.test.WebBrowserUtil$await$$anon$1@6f60bc0 (tried for 30 second(s) with 500 MILLISECONDS interval)
[info] Build info: version: '3.8.1', revision: '6e95a6684b', time: '2017-12-01T18:33:54.468Z'

[info] - with workspace defaults, with user preferences *** FAILED ***
[info]   org.openqa.selenium.TimeoutException: Expected condition failed: waiting for org.broadinstitute.dsde.workbench.service.test.WebBrowserUtil$await$$anon$1@4379eaeb (tried for 30 second(s) with 500 MILLISECONDS interval)
[info] Build info: version: '3.8.1', revision: '6e95a6684b', time: '2017-12-01T18:33:54.468Z'

[info] - with workspace defaults, with user preferences, new columns *** FAILED ***
[info]   org.openqa.selenium.TimeoutException: Expected condition failed: waiting for org.broadinstitute.dsde.workbench.service.test.WebBrowserUtil$await$$anon$1@6ed4245b (tried for 30 second(s) with 500 MILLISECONDS interval)
[info] Build info: version: '3.8.1', revision: '6e95a6684b', time: '2017-12-01T18:33:54.468Z'

[info] - keep ID column in download even if hidden in UI *** FAILED ***
[info]   org.openqa.selenium.TimeoutException: Expected condition failed: waiting for org.broadinstitute.dsde.workbench.service.test.WebBrowserUtil$await$$anon$1@53e90f64 (tried for 30 second(s) with 500 MILLISECONDS interval)
[info] Build info: version: '3.8.1', revision: '6e95a6684b', time: '2017-12-01T18:33:54.468Z'

[info] OrchestrationApiSpec:
[info] Orchestration
[info] - should not allow access alteration for arbitrary google roles *** FAILED ***
[info]   java.lang.Exception: Billing project creation did not complete
```

bah. let's just jenkins retest that again.

**Comment by davidangb** [comment]
https://fc-jenkins.dsp-techops.broadinstitute.org/job/firecloud-ui-fiab-test-runner/72/consoleFull

```
[info] *** 9 TESTS FAILED ***
[error] Failed: Total 116, Failed 9, Errors 0, Passed 107, Ignored 7
[error] Failed tests:
[error] 	org.broadinstitute.dsde.firecloud.test.security.AuthDomainSpec
[error] 	org.broadinstitute.dsde.firecloud.test.metadata.DataSpec
```

same failures in `DataSpec`, one new/different failure also in billing project creation:
```
[info] AuthDomainSpec:
[info] removing permissions from billing project owners for workspaces with auth domains
[info] - create workspace, + group member, + project owner, - project owner *** FAILED ***
[info]   java.lang.Exception: Billing project creation did not complete
```

File: src/main/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraService.scala [hunk 1]
```diff
@@ -1,15 +1,14 @@
 
 package org.broadinstitute.dsde.agora.server.webservice
 
-import akka.actor.Props
+import akka.actor.{ActorRef, Props}
```
Comment (rushtong) [inline src/main/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraService.scala]: `ActorRef` looks unused.

File: src/main/scala/org/broadinstitute/dsde/agora/server/webservice/ApiServiceActor.scala [hunk 2]
```diff
@@ -3,7 +3,7 @@ package org.broadinstitute.dsde.agora.server.webservice
 import java.security.Permissions
 
 import akka.actor.SupervisorStrategy.Restart
-import akka.actor.{OneForOneStrategy, Props}
+import akka.actor.{ActorRef, ActorRefFactory, OneForOneStrategy, Props}
```
Comment (rushtong) [inline src/main/scala/org/broadinstitute/dsde/agora/server/webservice/ApiServiceActor.scala]: `ActorRefFactory` looks unused

File: src/main/scala/org/broadinstitute/dsde/agora/server/webservice/ApiServiceActor.scala [hunk 3]
```diff
@@ -19,14 +19,17 @@ import spray.http.StatusCodes._
 import spray.routing._
 import spray.util.LoggingContext
 
+import scala.concurrent.ExecutionContext
```
Comment (rushtong) [inline src/main/scala/org/broadinstitute/dsde/agora/server/webservice/ApiServiceActor.scala]: `ExecutionContext` looks unused

**Review by rushtong — APPROVED** [review]
👍 Looks great! Tested locally with different conditions and it works as expected. Trivial comments inline that do not need re-review if addressed.

**Pull request merged**